jscomp/syntax/src/res_printer.ml

```diff
@@ -431,16 +431,18 @@ let classifyIdentContent ?(allowUident = false) txt =
         match String.unsafe_get txt i with
         | 'A' .. 'Z' when allowUident -> loop (i + 1)
         | 'a' .. 'z' | '_' -> loop (i + 1)
+        | '-' when allowHyphen -> loop (i + 1)
```

Comment:
here we make identifiers with hyphen regular identifier instead of exotic ones that need to be printed inside double quotes